docs/projections/user-defined-projections.md

```diff
@@ -0,0 +1,101 @@
+# User defined projections
+
+::: warning WIP
+This page is work in progress.
+:::
+
+<!-- TODO: Again refactor to shopping cart? -->
+
+<!-- TODO: Use cases and performance implications. -->
+
+<!-- ## Overview -->
+
+You create user defined projections in JavaScript. For example, the `my_demo_projection_result` projection below counts the number of `myEventType` events from the `account-1` stream. It then uses the `transformBy` function to change the final state:
+
+```javascript
+options({
+	resultStreamName: "my_demo_projection_result",
+	$includeLinks: false,
+	reorderEvents: false,
+	processingLag: 0
+})
+
+fromStream('account-1')
+.when({
+	$init:function(){
+		return {
+			count: 0
+		}
+	},
+	myEventType: function(state, event){
+		state.count += 1;
+	}
+})
+.transformBy(function(state){
+	state.count = 10;
+})
+.outputState()
+```
+
+<!-- TODO: Show example output, see above comment -->
+
+## User defined projections API
+
+### Options
+
+| Name | Description | Notes
+|:---- |:----------- |:-----
+`resultStreamName` | Overrides the default resulting stream name for the `outputState()` transformation, which is `$projections-{projection-name}-result`. | |
+| `$includeLinks` |Configures the projection to include/exclude link to events. | Default: `false` |
+| `processingLag` | When `reorderEvents` is enabled, this value is used to compare the total milliseconds between the first and last events in the buffer and if the value is equal or greater, the events in the buffer are processed. The buffer is an ordered list of events. | Default: `500ms`. Only valid for `fromStreams()` selector |
+| `reorderEvents` | Process events by storing a buffer of events ordered by their prepare position | Default: `false`. Only valid for `fromStreams()` selector |
+
+### Selectors
+
+| Selector | Description | Notes
+|:-------- |:----------- |:----- 
+| `fromAll()` | Selects events from the `$all` stream. | **Provides** <ul><li>`partitionBy`</li><li>`when`</li><li>`foreachStream`</li><li>`outputState`</li></ul> |
+| `fromCategory({category})` | Selects events from the `$ce-{category}` stream. | **Provides** <ul><li>`partitionBy`</li><li>`when`</li><li>`foreachStream`</li><li>`outputState`</li></ul> |
+| `fromStream({streamId})` | Selects events from the `streamId` stream. | **Provides** <ul><li>`partitionBy`</li><li>`when`</li><li>`outputState`</li></ul> |
+| `fromStreams(streams[])` | Selects events from the streams supplied. | **Provides**<ul><li>`partitionBy`</li><li>`when`</li><li>`outputState`</li></ul> |
+| `fromStreamsMatching(function filter)` | Selects events from the `$all` stream that returns true for the given filter. | **Provides** <ul><li>`when`</li></ul> |
```

Comment:
`fromStreamsMatching` has been removed as of 20.6.0

Reply:
Removed in https://github.com/EventStore/EventStore/pull/2831/commits/0c456d65092e8be301d5f3eb7e811c77a04417a9.